lib/spack/external/archspec/json/cpu/microarchitectures.json

```diff
@@ -2017,6 +2017,13 @@
       "features": [],
       "compilers": {
       }
+    },
+    "riscv64": {
+      "from": [],
+      "vendor": "generic",
+      "features": [],
+      "compilers": {
+      }
```

Comment:
Can you contribute this change to the upstream project at https://github.com/archspec/archspec-json ? 

What we usually do to extend microarchitecture support is to work on merging PRs in archspec and then update the version vendored with Spack. It would be good if you can also add some data to implement detection unit tests, like the one that is already present in https://github.com/archspec/archspec-json/tree/master/tests/targets for other microarchitectures

Reply:
Sure thing, I just did that here: https://github.com/archspec/archspec-json/pull/35

As part of the pull request, I added a uarch for SiFive u74-mc, which is the uarch of the board I'm using. I'm not entirely sure how to test archspec-json changes with my 'riscv' spack devel branch, so if you have any guidance there, would appreciate it. So far, the only thing I've done is copy the revised microarchitectures.json into my spack tree and tested that "spack install" still works.